src/manager/metrics.c

```diff
@@ -0,0 +1,140 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include "libhirte/common/common.h"
+#include "libhirte/common/parse-util.h"
+#include "libhirte/log/log.h"
+
+#include "job.h"
+#include "manager.h"
+#include "metrics.h"
+#include "node.h"
+
+static const sd_bus_vtable metrics_api_vtable[] = {
+        SD_BUS_VTABLE_START(0),
+
+        SD_BUS_SIGNAL_WITH_NAMES(
+                        "StartUnitJobMetrics",
+                        "ssstt",
+                        SD_BUS_PARAM(node_name) SD_BUS_PARAM(job_id) SD_BUS_PARAM(unit)
+                                        SD_BUS_PARAM(job_measured_time) SD_BUS_PARAM(unit_start_prop_time),
+                        0),
+        SD_BUS_SIGNAL_WITH_NAMES(
+                        "AgentJobMetrics",
+                        "ssst",
+                        SD_BUS_PARAM(node_name) SD_BUS_PARAM(unit) SD_BUS_PARAM(method)
+                                        SD_BUS_PARAM(systemd_job_time_micros),
+                        0),
+        SD_BUS_VTABLE_END
+};
+
+int metrics_api_service_register(Manager *manager) {
+        int r = sd_bus_add_object_vtable(
+                        manager->api_bus,
+                        &manager->metrics_slot,
+                        METRICS_OBJECT_PATH,
+                        METRICS_INTERFACE,
+                        metrics_api_vtable,
+                        manager);
```

Comment:
Instead of the manager, simply passing `NULL` seems more appropriate since the vtable only contains signals and does not need any userdata. I'd probably still keep the manager as input parameter to the function (although a ref to the bus would be sufficient) since this makes it more explicit that this function should only used for that.

Reply:
Ack